amm/interp/src/main/scala/ammonite/interp/AmmonitePlugin.scala

```diff
@@ -284,25 +284,30 @@ object LineNumberModifier {
                        topWrapperLen: => Int) = {
 
     object LineNumberCorrector extends g.Transformer {
+      import scala.reflect.internal.util._
+
+      private val trimmedSource = new BatchSourceFile(g.currentSource.file,
+        g.currentSource.content.drop(topWrapperLen))
+
       override def transform(tree: g.Tree) = {
         val transformedTree = super.transform(tree)
-        tree.pos match {
-          case s: scala.reflect.internal.util.OffsetPosition =>
-            if(s.point > topWrapperLen) {
-              val con = new BatchSourceFile(s.source.file,
-                s.source.content.drop(topWrapperLen))
-              val p = new OffsetPosition(con, s.point - topWrapperLen)
-              transformedTree.pos = p
-            }
-          case _ =>    //for position  = NoPosition
+        val newPos = tree.pos match {
+          case s : TransparentPosition if s.start > topWrapperLen =>
+              new TransparentPosition(trimmedSource, s.start, s.point - topWrapperLen, s.end)
+          case s: RangePosition if s.start > topWrapperLen =>
+              new RangePosition(trimmedSource, s.start, s.point - topWrapperLen, s.end)
+          case s: OffsetPosition if s.start > topWrapperLen =>
```

Comment:
s.start == s.point in this case

Reply:
Shouldn't the `startIn` of the `RangePosition` also be `s.start - topWrapperLen`? and the `endIn` be `s.end - topWrapperLen`?

Reply:
Ideally it would be the case, but the compiler performs some later validation on the trees to make sure the ranges aren't overlapping, which throws when I try what your suggest. I'm not quite sure why, it could have to do with multi-phase scripts. 

With the current change, the behaviour of ammonite remains the same whilst allowing external tools (semanticdb) to work around this issue by adding post-processing logic that would look up the position of the symbol matching the filename (thanks for that btw 😄) and applying the appropriate shift. 

I can open a ticket for further investigation and reference it in a TODO in this piece of code, if that is acceptable

Reply:
There is a ticket to tone down position validation errors in the compiler https://github.com/scala/scala-dev/issues/390 The validation is required for Scala IDE to efficiently find enclosing trees given a position. semanticdb does not require the validated properties to hold so the errors are unnecessary.

Reply:
Sure that sounds good to me

Reply:
Done !  fb8ed68